### PR TITLE
Ca feature/reporter base

### PR DESCRIPTION
**Pull request by cagodoy**
In the following PR the following has been implemented:
- commander to run cronjob
- internal pkg to get status from synchronizers
- makefile
- docker file for image
- docker compose for development

### CODE REVIEW

File: cmd/reporter/main.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	// check serviceType is valid
+	st := reporter.ServiceType(serviceType)
+
+	// initialize storage
+	s, err := storage.New(databaseURL)
+	if err != nil {
+		log.Fatal(err)
+	}
```
Comment (mtavano) [inline cmd/reporter/main.go]: please, use config structs instead use a bunch of lines for this.

File: cmd/reporter/main.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	// fetch to jobs
+
+	// fetch to nodes
+
+	// initilize struct
+	// - add array with len of events
+	// for i, e := 
+	// groupReport := &reporter.GroupReport{
+	// 	Synchronizers: &reporter.Report{
```
Comment (mtavano) [inline cmd/reporter/main.go]: delete this comments

File: internal/commander/commander.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	if (t == reporter.ServiceTypeJobs) {
+		return jobs()
+	}
+
+	if (t == reporter.ServiceTypeNodes) {
+		return nodes()
+	}
+
+	return fmt.Errorf("invalid service_type=%s in Reporter(t)", string(t))
+}
```
Comment (mtavano) [inline internal/commander/commander.go:27]: cant get the idea behind this. what if this method for?

File: internal/storage/reporter/reporter.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	// parse strut to bytes
+	b, err := json.Marshal(g)
+	if err != nil {
+		return err
+	}
+
+	// save in database
+	err = s.storage.DB.Set(ctx, key, []byte(b), 0).Err()
+	if err != nil {
+		fmt.Println(9999)
```
Comment (mtavano) [inline internal/storage/reporter/reporter.go]: drop this Println

File: internal/storage/reporter/reporter.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+func (s *Storage) GetGroupReport(key int64, t reporter.ServiceType) (*reporter.GroupReport, error) {
+	ctx := context.Background()
+
+	// format the composed key used in db
+	parsedKey := fmt.Sprintf("%s:%s:%v", prefix, t, key)
+
+	// get bytes by composed key to db
+	str, err :=	s.storage.DB.Get(ctx, parsedKey).Result()
+	if err == redis.Nil {
+		return nil, errors.New("key does not exist")
```
Comment (mtavano) [inline internal/storage/reporter/reporter.go:64]: Nil is a specific error of redis? maybe we should user `errors.Is` method

File: internal/storage/reporter/reporter.go [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+func (s *Storage) ListGroupReports(t reporter.ServiceType) ([]*reporter.GroupReport, error) {
+	ctx := context.Background()
+	
+	// prepare slice of group reports
+	groupReports := make([]*reporter.GroupReport, 0)
+
+	// // iterte over db elements and push in slice
+	iter := s.storage.DB.Scan(ctx, 0, fmt.Sprintf("%s:%s:*", prefix, t), 0).Iterator()
+	for iter.Next(ctx) {
+		fmt.Println("JAJAJAA keys", iter.Val())
```
Comment (mtavano) [inline internal/storage/reporter/reporter.go]: drop this println

File: internal/storage/reporter/reporter.go [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+		if err != nil {
+			return nil, err
+		}
+
+		// // parse bytes to group report struct
+		// var groupReport *reporter.GroupReport
+		// err = json.Unmarshal(gp, &groupReport)
+		// if err != nil {
+		// 	return nil, err
+		// }
```
Comment (mtavano) [inline internal/storage/reporter/reporter.go]: drop this comments

File: internal/storage/reporter/reporter.go [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+		// append new element to group reports slice
+		groupReports = append(groupReports, groupReport)
+	}
+	if err := iter.Err(); err != nil {
+		return nil, err
+	}
```
Comment (mtavano) [inline internal/storage/reporter/reporter.go:109]: isn't it better to evaluate this first on the loop?

File: internal/storage/storage.go [hunk 9]
```diff
@@ -0,0 +1,18 @@
+package storage
+
+import "github.com/go-redis/redis/v9"
+
+type S struct {
```
Comment (mtavano) [inline internal/storage/storage.go]: better names for structs?

File: cmd/reporter/main.go (hunk 2 above)
Comment (cagodoy) [reply cmd/reporter/main.go]: ok

File: cmd/reporter/main.go (hunk 1 above)
Comment (cagodoy) [reply cmd/reporter/main.go]: ok

File: internal/commander/commander.go (hunk 3 above)
Comment (cagodoy) [reply internal/commander/commander.go:27]: its a method to abstract the logic for getting status from the needed services, ex: synchronizer, jobs and nodes

File: internal/storage/reporter/reporter.go (hunk 4 above)
Comment (cagodoy) [reply internal/storage/reporter/reporter.go]: ok

File: internal/storage/reporter/reporter.go (hunk 6 above)
Comment (cagodoy) [reply internal/storage/reporter/reporter.go]: ok

File: internal/storage/reporter/reporter.go (hunk 7 above)
Comment (cagodoy) [reply internal/storage/reporter/reporter.go]: ok

File: internal/storage/reporter/reporter.go (hunk 5 above)
Comment (cagodoy) [reply internal/storage/reporter/reporter.go:64]: its necessary for to use Redis

File: internal/storage/storage.go (hunk 9 above)
Comment (cagodoy) [reply internal/storage/storage.go]: ok

**Pull request merged**